airflow/contrib/operators/gcp_function_operator.py

```diff
@@ -0,0 +1,571 @@
+
+    def _validate_union(self, children_validation_specs, full_field_path,
+                        dictionary_to_validate):
+        # type: (dict, str, dict) -> None
+        field_found = False
+        found_field_name = None
+        for child_validation_spec in children_validation_specs:
+            # Forcing optional so that we do not have to type optional = True
+            # in specification for all union fields
+            new_field_found = self._validate_field(
+                validation_spec=child_validation_spec,
+                dictionary_to_validate=dictionary_to_validate,
+                parent=full_field_path,
+                force_optional=True)
+            field_name = child_validation_spec['name']
+            if new_field_found and field_found:
+                raise FieldValidationException(
+                    "The field '{}' and '{}' are part of '{}' union so they are mutually "
+                    "exclusive. They should not be both present in the same dictionary!".
+                    format(field_name, found_field_name, full_field_path))
+            if new_field_found:
+                field_found = True
+                found_field_name = field_name
+        if not field_found:
+            self.log.warning(
+                "None of available variants of '{}' union was found in dictionary {}!. "
+                "Should be one of {}".format(
+                    full_field_path,
+                    dictionary_to_validate,
+                    [field['name'] for field in children_validation_specs]))
+
+    def _validate_field(self, validation_spec, dictionary_to_validate, parent=None,
+                        force_optional=False):
+        """
+        Validates if field is OK.
+        :param validation_spec: specification of the field
+        :type validation_spec: dict
+        :param dictionary_to_validate: dictionary where the field should be present
+        :type dictionary_to_validate: dict
+        :param parent: full path of parent field
+        :type parent: str
+        :param force_optional: forces the field to be optional
+        (all union fields have force_optional set to True)
+        :type force_optional: bool
+        :return: True if the field is present
+        """
+        field_name = validation_spec['name']
+        field_type = validation_spec.get('type')
+        optional = validation_spec.get('optional')
+        regexp = validation_spec.get('regexp')
+        children_validation_specs = validation_spec.get('fields')
+        required_api = validation_spec.get('api')
+        custom_validation = validation_spec.get('custom_validation')
+
+        full_field_path = self._get_field_name_with_parent(field_name=field_name,
+                                                           parent=parent)
+        if required_api and required_api != self._api:
+            self.log.debug(
+                "Not validating the field '{}' for api version '{}' "
```

Comment:
Nit: capitalize API.

Reply:
fixed in 3722a2be480c6d8080fc60491e8fa431b7eb6590